runtime/common/src/migrations/balances.rs

```diff
@@ -0,0 +1,301 @@
+	pub fee_frozen: Balance,
+}
+
+pub type OldAccountInfoOf<T> = AccountInfo<
+	<T as frame_system::Config>::Index,
+	OldAccountData<<T as pallet_balances::Config>::Balance>,
+>;
+
+pub type NewAccountData<Balance> = pallet_balances::AccountData<Balance>;
+
+pub type NewAccountInfoOf<T> = AccountInfo<
+	<T as frame_system::Config>::Index,
+	NewAccountData<<T as pallet_balances::Config>::Balance>,
+>;
+
+pub struct Migration<T: pallet_balances::Config + frame_system::Config>(
+	sp_std::marker::PhantomData<T>,
+);
+
+impl<T> OnRuntimeUpgrade for Migration<T>
+where
+	T: frame_system::Config<AccountId = AccountId32, Index = u32>
+		+ pallet_balances::Config<Balance = u128>,
+	NewAccountInfoOf<T>:
+		codec::EncodeLike<AccountInfo<u32, <T as frame_system::Config>::AccountData>>,
+{
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+		check_account_storage::<T>(true);
+
+		Ok(vec![])
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		// !!! REMOVE TEST ACCOUNT DATA !!!
+		//
+		// TEST ACCOUNT DATA - START
+		let test_account_data = get_test_account_data::<T>();
+
+		for (account_id, account_data) in test_account_data {
+			log::info!(
+				"Balances Migration - Processing account id - {}",
+				hex::encode(account_id.clone())
+			);
+
+			frame_system::Account::<T>::insert(account_id, account_data);
+		}
+
+		// TEST ACCOUNT DATA - END
+
+		// WE CAN NOT MIGRATE. THIS CODE IS JUST FOR CHECKING IF WE NEED ANYTHING
+		// BESIDES THE LAZY MIGRATION FROM PARITY
+		// See: https://kflabs.slack.com/archives/C05TBFRBL15/p1699956615421249
+		Weight::from_parts(0, 0)
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(_: Vec<u8>) -> Result<(), DispatchError> {
+		check_account_storage::<T>(false);
+
+		Ok(())
+	}
+}
+
+/// The difference between the old and new account data structures
+/// is the last field of the struct, which is:
+//
+/// * Old - `free_frozen` - T::Balance
+/// * New - `flags` - u128
+//
+/// During this check, we confirm that the old or the new AccountInfo can
+/// be successfully decoded given the raw data found in storage, and
+/// we add specific checks for each version:
+//
+/// * Old - confirm that `fee_frozen` is zero, as it shouldn't have been used so
+///   far
+/// * New - confirm that `flags` does not have the new logic flag set because
+///   this requires calling
+///   `pallet_balances::Pallet::<T>::upgrade_account(origin, &who)` which can be
+///   done by a script
+#[cfg(feature = "try-runtime")]
+fn check_account_storage<T: frame_system::Config + pallet_balances::Config>(is_old: bool) {
+	let account_prefix = frame_system::Account::<T>::final_prefix();
+
+	let mut total_count = 0;
+	let mut total_reserved = 0;
+	let mut total_frozen = 0;
+	let mut total_misc_frozen = 0;
+	let mut total_flags = 0;
+
+	let prefix = account_prefix.to_vec();
+	let mut previous_key = prefix.clone();
+
+	while let Some(next) = sp_io::storage::next_key(&previous_key) {
+		if !next.starts_with(&prefix) {
```

Comment:
Great catch! I also had this initially but removed it since I was trying to use the `pallet_balances::Account` prefix and wasn't getting any results... fun times.